### PR TITLE
Optimize if/select when condition not equal to 64-bit zero

### PR DESCRIPTION
**Pull request by MaxGraey**
`(int64)x != 0 ? y : z` ==> `!x ? z : y`

### CODE REVIEW

File: src/passes/OptimizeInstructions.cpp [hunk 1]
```diff
@@ -767,6 +767,18 @@ struct OptimizeInstructions
     } else if (auto* iff = curr->dynCast<If>()) {
       iff->condition = optimizeBoolean(iff->condition);
       if (iff->ifFalse) {
+        if (auto* binary = iff->condition->dynCast<Binary>()) {
+          if (binary->op == NeInt64) {
+            if (auto* c = binary->right->dynCast<Const>()) {
+              if (c->value.geti64() == 0LL) {
+                // if((int64)x != 0) y else z  ==>  if(!x) z else y
```
Comment (kripken) [inline src/passes/OptimizeInstructions.cpp:774]: is this an i64 version of something we have an i32 optimization for? if so could this be alongside it, or even merged with it using Abstract etc.?

Comment (MaxGraey) [reply src/passes/OptimizeInstructions.cpp:774]: No, it's special only for 64-bit. For 32-bits it's already handled in `optimizeBoolean` and [slightly differently](https://github.com/WebAssembly/binaryen/blob/765c295fda721296637b841201c057c69a579a4d/src/passes/OptimizeInstructions.cpp#L1021)

Comment (MaxGraey) [reply src/passes/OptimizeInstructions.cpp:774]: Added comment about this

**Comment by MaxGraey** [comment]
It seems CI is broken: https://github.com/WebAssembly/binaryen/pull/2931/checks?check_run_id=813584426#step:6:201

> Error: Cannot find module 'acorn'

**Comment by MaxGraey** [comment]
Closing in favour #4194

**Pull request closed without merging**